### ISSUE TITLE
[Bug]: get_shape() core dump

### ISSUE THREAD

**Issue by LinGeLin** [openvinotoolkit/openvino#26941]
### OpenVINO Version

2024.4.0

### Operating System

Ubuntu 20.04 (LTS)

### Device used for inference

CPU

### Framework

Keras (TensorFlow 2)

### Model used

ps model

### Issue description

I am modifying a project, changing the tf infer to use ov infer. This is a C++ project, and there are some tensor variables inside. I have replaced them all with ov::Tensor. After the modifications, when running the project, some tensors cause a coredump when executing get_shape. However, the location where the core dump occurs seems strange to me. In this line, there is a check to see if _impl is nullptr, so it should not cause a core dump in theory. What are some troubleshooting ideas?

`
#0  0x00007f131f1a538f in ov::Tensor::get_shape (this=0x166a0170)
    at /data/xxx/workspace/openvino/src/core/src/runtime/tensor.cpp:87
87	    OV_TENSOR_STATEMENT(return _impl->get_shape());
`

### Issue submission checklist

- [X] I'm reporting an issue. It's not a question.
- [X] I checked the problem with the documentation, FAQ, open issues, Stack Overflow, etc., and have not found a solution.
- [X] There is reproducer code and related data files such as images, videos, models, etc.

**Comment by mlukasze** [comment]
@praasz please evaluate
In case bug will be confirmed please produce ticket and let's discuss schedule internally

**Comment by rkazants** [comment]
Hi @LinGeLin, can you send more details pls (like code how you use OpenVINO)?

Best regards,
Roman

**Comment by LinGeLin** [comment]
> Hi @LinGeLin, can you send more details pls (like code how you use OpenVINO)?
> 
> Best regards, Roman

It goes something like this: I first allocate inputTensor for my model and save it as a member variable of a class, and based on this class, I create a memory pool. I just rechecked the project again, and I found that I missed modifying some code. It is likely that this part of the code caused memory overwriting when accessing inputTensor, making inputTensor invalid. I am very sorry; I should have carefully rechecked the code before asking the question.
